Realm/Tests/RealmTests.mm

```diff
@@ -1907,6 +1909,68 @@ - (void)testReadAfterInvalidateFrozen {
                               @"Cannot read from a frozen Realm which has been invalidated.");
 }
 
+- (void)testThaw {
+    RLMRealm *realm = [RLMRealm defaultRealm];
+    XCTAssertFalse(realm.frozen);
+
+    [realm beginWriteTransaction];
+    [IntObject createInRealm: realm withValue:@[@1]];
+    [realm commitWriteTransaction];
+
+    RLMRealm *frozenRealm = [realm freeze];
+    XCTAssertTrue(frozenRealm.frozen);
+    IntObject *frozenObj = [[IntObject objectsInRealm:frozenRealm where:@"intCol = 1"] firstObject];
+    XCTAssertTrue(frozenObj.frozen);
+
+    RLMRealm *thawedRealm = [realm thaw];
+    XCTAssertFalse(thawedRealm.frozen);
+    IntObject *thawedObj = [[IntObject objectsInRealm:thawedRealm where:@"intCol = 1"] firstObject];
+
+    [realm beginWriteTransaction];
+    thawedObj.intCol = 2;
+    [realm commitWriteTransaction];
+    XCTAssertNotEqual(thawedObj.intCol, frozenObj.intCol);
+
+    IntObject *nilObj = [[IntObject objectsInRealm:thawedRealm where:@"intCol = 1"] firstObject];
+    XCTAssertNil(nilObj);
+}
+
+- (void)testThawDifferentThread {
+    RLMRealm *frozenRealm = [[RLMRealm defaultRealm] freeze];
+    XCTAssertTrue(frozenRealm.frozen);
+    
+    // Thaw on a thread which already has a Realm should use existing reference.
+    [self dispatchAsyncAndWait:^{
```

Comment:
@tgoyne 

> Thaw on a thread which already has a Realm should use that realm instance
> Thaw on a thread which doesn't already have a Realm instance

Do these tests demonstrate what you mean? ie. thaw isn't creating a bunch of references when one already exists?
<img width="643" alt="Screen Shot 2020-12-17 at 6 23 28 PM" src="https://user-images.githubusercontent.com/25092858/102555541-ff730e80-4094-11eb-8ea4-7b45c80866c4.png">